content/posts/android-tv-12.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Android TV v0.12
+subtitle: Our biggest app update to date.
+author: Niels van Velzen
+githubusername: nielsvanvelzen
+date: 2021-09-30
+justify: center
+---
+
+<!-- markdownlint-disable MD033 MD036 -->
+
+Today we're bringing the Android TV app to 2021 with a modern design and tons of other changes.
+
+<!--more-->
+
+Over 400 [^project] pull requests containing 2400+ commits, 750+ changed files with 54000+ changed lines of code by roughly 50 contributors[^git-diff]. This is the biggest update to the app we ever worked on. I'll talk a bit about why it is this big and why it took so long later in the post. But let's start with the part you probably came for: **new features!**
+
+[^project]: [v0.12.0 project filtered by merged pull requests](https://github.com/jellyfin/jellyfin-androidtv/projects/2?card_filter_query=is%3Apr+is%3Amerged)
+[^git-diff]: [Git comparison between v0.11.5 and v0.12.0-beta.7](https://github.com/jellyfin/jellyfin-androidtv/compare/v0.11.5...v0.12.0-beta.7)
+
+## New authentication flow
+
+We got a lot of complaints about our authentication flow. It wasn't obvious how to enable the auto log in option and managing multiple servers or users was not an easy task. We also didn't like this part of the app, and decided to completely revamp it! The rewritten sign in screen looks more modern and is much easier to use. There is a new help section that links to our documentation to help new users getting started with Jellyfin. The auto-discovery feature of the app now shows all servers instead of the first one, you can select them to connect or manually enter your server address. You can then proceed to add a user and start using the app. Users are automatically saved now with auto log in enabled by default.
```

Comment:
```suggestion
We got a lot of complaints about our authentication flow. It wasn't obvious how to enable the auto-login option and managing multiple servers or users was not an easy task. We also didn't like this part of the app, and decided to completely revamp it! The rewritten sign-in screen looks more modern and is much easier to use. There is a new help section that links to our documentation to help new users get started with Jellyfin. The auto-discovery feature of the app now shows all servers instead of the first one, you can select one of those or manually enter your server address to connect . You can then proceed to add a user and start using the app. Users are automatically saved now with auto-login enabled by default.
```